drivers/spi/spi_ll_stm32.c

```diff
@@ -326,6 +326,39 @@ static void spi_stm32_read_next_frame(SPI_TypeDef *spi,
 	}
 }
 
+#if SPI_STM32_FIFO_ENABLED
+static void spi_stm32_send_next_packet(SPI_TypeDef *spi,
+		struct spi_stm32_data *data)
+{
+	uint8_t packet_size = SPI_PACKET_SIZE_GET(data->ctx.config->operation);
```

Comment:
Packet is really not relevant in SPI. Packet holds a meaning about a start and an end of data (thus its usage in network for instance)
In SPI, having a fifo in the controller does not mean you are going to send/receive "packets". 

You are just feeding/reading a fifo according to a threshold. Imo, it would more relevant to use that word.
You would just need to loop on sending/reading around data->threshold

Reply:
Given that this is a driver for stm32, I opted for using the nomenclature that stm32 already uses, so potential users understand it better as the code refers to the same concepts as the stm32 datasheet and HAL library. 

E.g. see the datasheet capture below relating the FIFO threshold with packet size:

![ifoth](https://github.com/zephyrproject-rtos/zephyr/assets/127399215/5037890b-7dae-4307-8d38-d5d99003769f)


Also, see the following function generated by STM32CubeMX:

```
/**
  * @brief  Compute configured packet size from fifo perspective.
  * @param  hspi: pointer to a SPI_HandleTypeDef structure that contains
  *               the configuration information for SPI module.
  * @retval Packet size occupied in the fifo
  */
static uint32_t SPI_GetPacketSize(SPI_HandleTypeDef *hspi)
{
  uint32_t fifo_threashold = (hspi->Init.FifoThreshold >> SPI_CFG1_FTHLV_Pos) + 1UL;
  uint32_t data_size       = (hspi->Init.DataSize      >> SPI_CFG1_DSIZE_Pos) + 1UL;

  /* Convert data size to Byte */
  data_size = (data_size + 7UL) / 8UL;

  return data_size * fifo_threashold;
}

```

This argument is applicable only to the stm32 SPI driver, not to the `spi.h` Zephyr interface.

Reply:
weird naming choice from ST then :)

Reply:
Solved as this name is no longer used.

Reply:
@teburd @tbursztyka unless you have any objection, I will resolve this conversation.